### PR TITLE
azure-core 1.5 fixed this bug on odata.type

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**

### CODE REVIEW

File: sdk/monitor/mgmt/src/main/java/com/azure/management/monitor/Action.java [hunk 1]
```diff
@@ -14,7 +15,7 @@
 @JsonTypeInfo(
     use = JsonTypeInfo.Id.NAME,
     include = JsonTypeInfo.As.PROPERTY,
-    property = "odata.type",
+    property = "odata\\.type",
```
Comment (xccc-msft) [inline sdk/monitor/mgmt/src/main/java/com/azure/management/monitor/Action.java:18]: Did you try to run living test for monitor locally after these changes?

Comment (weidongxu-microsoft) [reply sdk/monitor/mgmt/src/main/java/com/azure/management/monitor/Action.java:18]: Already had a test for this.
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/monitor/mgmt/src/test/java/com/azure/management/monitor/TypeSerializationTests.java#L19

**Comment by yungezz** [comment]
hi @weidongxu-microsoft could you pls share more context about the change such as issue link etc? thanks.

**Comment by weidongxu-microsoft** [comment]
@yungezz 

Issue on azure-core https://github.com/Azure/azure-sdk-for-java/issues/10217, which is fixed in 1.5.0

**Pull request merged**